manage/upgrading/version_specific_migration/upgrade_to_52.rst

```diff
@@ -5,28 +5,315 @@ Upgrading Plone 5.1 to 5.2
+
+.. code-block:: python
+
+    # old:
+    import Globals
+    develoment_mode = Globals.DevelopmentMode
+
+    # new
+    from App.config import getConfiguration
+    develoment_mode = getConfiguration().debug_mode
+
+Functional tests using the zope.testbrowser now use ``WebTest`` instead of ``mechanize``. That means that tests that used interal methods of mechanize need to be updated.
+
+
+WSGI
+----
+
+This is a result of the PLIP for Python 3. Plone 5.2 by default uses the WSGI-Server ``waitress`` to
+
+For End Users
+~~~~~~~~~~~~~
+
+Nothing changes.
+
+For Developers
+~~~~~~~~~~~~~~
+
+By default Plone uses ``waitress`` instead of ``ZServer`` as a http-server since ``ZServer`` will not ported to Python 3. Only when running on Python 2 you can still decide to use ``ZServer`` by setting ``wsgi = off`` in the buildout-part that configures the instance with ``plone.recipe.zope2instance``.
+
+Some options that used to configure ``ZServer`` are no longer available in ``plone.recipe.zope2instance`` when running on ``WSGI``. Check https://pypi.org/project/plone.recipe.zope2instance for details.
+
+
+plone.restapi
+-------------
+
+This is `PLIP 2177 <https://github.com/plone/Products.CMFPlone/issues/2177>`_.
+
+For End Users
+~~~~~~~~~~~~~
+
+Nothing changes.
+
+For Developers
+~~~~~~~~~~~~~~
+
+You can now use a RESTful hypermedia API for Plone to build modern JavaScript front-ends on top of Plone.
+
+See https://plonerestapi.readthedocs.io/en/latest/ for details.
+
+
+New navigation with dropdown
+----------------------------
+
+This is `PLIP 2516 <https://github.com/plone/Products.CMFPlone/issues/2516>`_.
+
+
+For End Users
+~~~~~~~~~~~~~
+
+Site-Administrators can use the navigation controlpanel (``/@@navigation-controlpanel``) to configure the dropdown-navigation.
+
+
+For Developers
+~~~~~~~~~~~~~~
+
+For upgraded sites the dropdown-navigation is disabled by default, for new sites it is set to display 3 levels.
+
+The code for the global navigation has moved to ``plone.app.layout.navigation.navtree.NavTreeProvider`` and the template ``plone.app.layout/plone/app/layout/viewlets/sections.pt`` has changed. Overrides of the previous navigation may no longer work and need to be updated.
+
+Developers who used add-ons or custom code for a dropdown-navigation should consider migrating to the new navigation since it is extremely fast, accessible and implemented almost entirely with css and html.
+
+
+Merge Products.RedirectionTool into core
+----------------------------------------
+
+This is `PLIP 1486 <https://github.com/plone/Products.CMFPlone/issues/1486>`_.
+
+For End Users
+~~~~~~~~~~~~~
+
+Site-Administrators can use the URL Management control panel (``/@@redirection-controlpanel``) to manage and add alternative URLs.
+
+
+For Developers
+~~~~~~~~~~~~~~
+
+Since the add-on ``Products.RedirectionTool`` has been merged into Plone, you should remove it before the upgrade if you previously used it.
+
```

Comment:
Maybe I am stupid, but currently, I can not find any docs about ``Products.RedirectionTool`` in the docs branch 5.2.
We should make sure that we add these, I guess if I remember right there is also some UI involved, meaning pictures could not hurt :)

Reply:
This feature is visible as a new controlpanel that needs to be documented. The name ``Products.RedirectionTool`` should be mentioned but is no longer used. I think
@staeff and @mauritsvanrees did most of the work. Maybe they could document it.

Reply:
For documentation, see my [comment in the PLIP(https://github.com/plone/Products.CMFPlone/issues/1486#issuecomment-461637558) mentioning @svx. For now, that commit is the documentation. :-)

Reply:
I added some of the text here.